### PR TITLE
[spaceship][fastlane_core] run iTMSTransporter with `xcrun` if Xcode 11 or up

### PR DESCRIPTION
**Pull request by joshdholtz**
# Motivation and Context
Fixes https://github.com/fastlane/fastlane/issues/15323

- Started receiving error of `Transporter error: Could not create the Java Virtual Machine`

### Description
- use `xcrun iTMSTransporter` if Xcode 11 or up
- fixes issues that if not using correct version of Java which `iTMSTransporter` needs (I think)
  - `xcrun` is in charge of the environment

### CODE REVIEW

**Comment by joshdholtz** [comment]
@SConaway This should fix your error that you mentioned here 👉 https://github.com/fastlane/fastlane/issues/16639#issuecomment-649805838

**Comment by Niek** [comment]
Thanks a lot, this fixed the issues I was having on Xcode 11 and 12 beta.

**Comment by joshdholtz** [comment]
@Niek Awesome! Thanks for testing that out 😊 My apologies on missing that original issue and not getting this fixed sooner!

**Comment by SConaway** [comment]
Cool. Has this made it into an `rc` release?

File: fastlane_core/lib/fastlane_core/itunes_transporter.rb [hunk 1]
```diff
@@ -239,64 +239,101 @@ def shell_escaped_password(password)
...
+        [
+          'xcrun iTMSTransporter',
+          '-m upload',
+          "-u #{username.shellescape}",
+          "-p #{password.shellescape}",
+          "-f #{source.shellescape}",
+          additional_upload_parameters, # that's here, because the user might overwrite the -t option
+          '-k 100000',
+          ("-itc_provider #{provider_short_name}" unless provider_short_name.to_s.empty?),
+          '2>&1' # cause stderr to be written to stdout
```
Comment (minuscorp) [inline fastlane_core/lib/fastlane_core/itunes_transporter.rb:252]: I'm wondering if we can get separate pipes for stdout and stderr and then treat each one individually. What do you think?

**Review by janpio — APPROVED** [review]
so much cleaner

File: fastlane_core/lib/fastlane_core/itunes_transporter.rb (hunk 1 above)
Comment (joshdholtz) [reply fastlane_core/lib/fastlane_core/itunes_transporter.rb:252]: I’m very hesitant to change that right now 😇 I’d be down for experimenting in a separate PR after 2.150.0 😊 I just want to fix things to working lol

**Comment by changusmc** [comment]
@joshdholtz I had an issue with this change where we were getting an error.

```
Jun 30 08:30:32.555: [01:30:32]: Login to App Store Connect (ios-jenkins@dropbox.com)
Jun 30 08:30:35.457: [01:30:35]: Login successful
Jun 30 08:30:41.696: [01:30:41]: Ready to upload new build to TestFlight (App: 327630330)...
Jun 30 08:30:43.006: [01:30:42]: Going to upload updated app to App Store Connect
Jun 30 08:30:43.006: [01:30:42]: This might take a few minutes. Please don't interrupt the script.
Jun 30 08:30:57.753: [01:30:57]: [Transporter Error Output]: Your account information was entered incorrectly. (-20101)
Jun 30 08:30:57.861: [01:30:57]: Transporter transfer failed.
```

When reverting to rc3, we didn't see this error.

We ended up changing our password to not contain certain special characters. You might need to escape them.

**Pull request merged**